app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -153,9 +198,27 @@ export class CognitoImpl implements Cognito {
 
     // === Interface `impl`s ===
 
+    /** We want to signal to Amplify to fire a "custom state change" event when the user is
+     * redirected back to the application after signing in via an external identity provider. This
+     * is done so we get a chance to fix the location history that Amplify messes up when it
+     * redirects the user to the identity provider's authentication page.
```

Comment:
What does it mean that "Amplify messes up the location history"? A small explanation of what that means should be provided.

Reply:
Updated the docs.